generator/internal/stub_generator.cc

```diff
@@ -211,33 +216,62 @@ Status StubGenerator::GenerateHeader() {
     "\n"
     "class Default$stub_class_name$ : public $stub_class_name$ {\n"
     " public:");
+  std::unordered_map<std::string, std::string> mixin_grpc_stubs;
+  for (auto const& mixin_method : MixinMethods()) {
+    mixin_grpc_stubs[mixin_method.grpc_stub_name] = mixin_method.grpc_stub_fqn;
+  }
+
+  std::vector<std::string> mixin_stub_inputs_list;
+  std::vector<std::string> mixin_stub_members_init_list;
+  std::vector<std::string> mixin_stub_members_list;
+  for (auto const& mixin_grpc_stub : mixin_grpc_stubs) {
```

Comment:
I think it would be simpler to just loop over `MixinMethods()` and print individual lines of code where we need them. Then we will not need these auxiliary data structures.

Reply:
Do you mean I can do it using multiple `HeaderPrint`, like 
```
HeaderPrint(
R"""(
  Default$stub_class_name$(
      std::unique_ptr<$grpc_stub_fqn$::StubInterface> grpc_stub,
  )"""
)

for (auto const& method: mixinmethods()) {
   ... HeaderPrint ...
}

HeaderPrint(R"""(    std::unique_ptr<google::longrunning::Operations::StubInterface> operations)
      : grpc_stub_(std::move(grpc_stub)),
)""")

for (auto const& method: mixinmethods()) {
   ... HeaderPrint ...
}
```

Reply:
Yeah, basically. You definitely don't want to use `absl::StrJoin` because it forces you to do special handling of the empty case:

https://github.com/googleapis/google-cloud-cpp/blob/f0e7bd56e0aa82469ae3ef3a3c347bb57d8c1e01/generator/internal/stub_generator.cc#L245

It'll definitely be simpler if we append `"\n"` to every line of code.

Reply:
For these member variable initializations specifically, I would consider printing lines like:

```
, member_variable_(value)
```

The golden-files will not look exactly like what `clang-format` wants, but that is fine. The library files will get reformatted. It might make the handling cleaner.

Reply:
Done.